monkey/infection_monkey/post_breach/actions/communicate_as_new_user.py

```diff
@@ -38,95 +33,27 @@ def __init__(self):
-                        None,  # Process attributes
-                        None,  # Thread attributes
-                        True,  # Should inherit handles
-                        win32con.NORMAL_PRIORITY_CLASS,  # The priority class and the creation of the process.
-                        None,  # An environment block for the new process. If this parameter is NULL, the new process
-                        # uses the environment of the calling process.
-                        None,  # CWD. If this parameter is NULL, the new process will have the same current drive and
-                        # directory as the calling process.
-                        win32process.STARTUPINFO()  # STARTUPINFO structure.
-                        # https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/ns-processthreadsapi-startupinfoa
-                    )
-
-                    logger.debug(
-                        "Waiting for ping process to finish. Timeout: {}ms".format(PING_WAIT_TIMEOUT_IN_MILLISECONDS))
-
-                    # Ignoring return code, as we'll use `GetExitCode` to determine the state of the process later.
-                    _ = win32event.WaitForSingleObject(  # Waits until the specified object is signaled, or time-out.
-                        process_handle,  # Ping process handle
-                        PING_WAIT_TIMEOUT_IN_MILLISECONDS  # Timeout in milliseconds
-                    )
-
-                    ping_exit_code = win32process.GetExitCodeProcess(process_handle)
-
-                    self.send_ping_result_telemetry(ping_exit_code, commandline, username)
-                except Exception as e:
-                    # If failed on 1314, it's possible to try to elevate the rights of the current user with the
-                    #  "Replace a process level token" right, using Local Security Policy editing.
-                    PostBreachTelem(self, (
-                        "Failed to open process as user {}. Error: {}".format(username, str(e)), False)).send()
-                finally:
-                    try:
-                        win32api.CloseHandle(process_handle)
-                        win32api.CloseHandle(thread_handle)
-                    except Exception as err:
-                        logger.error("Close handle error: " + str(err))
-        except subprocess.CalledProcessError as err:
-            PostBreachTelem(self, (
-                "Couldn't create the user '{}'. Error output is: '{}'".format(username, str(err)),
-                False)).send()
         except NewUserError as e:
             PostBreachTelem(self, (str(e), False)).send()
 
+    @staticmethod
+    def get_random_new_user_name():
+        return USERNAME_PREFIX + ''.join(random.choice(string.ascii_lowercase) for _ in range(5))
+
+    @staticmethod
+    def get_commandline_for_ping(domain=PING_TEST_DOMAIN, is_windows=is_windows_os()):
+        if is_windows:
```

Comment:
It doesn't make sense that on windows `PING.exe`, `-n`, `1` are parameters, and on linux they're part of the format string.
Choose one style and be consistent. I would go with the linux style

Reply:
Once you do that, you can do something like:
`format_str = 'win f-str' if is_windows else 'linux f-str'`
`return format_str.format(domain)`